budgetportal/tests/test_prov_infra_projects.py

```diff
@@ -252,14 +533,175 @@ def test_url_path(self):
+
+        self.assertEqual(self.project_snapshot_2, latest)
+
+
+class ProvInfraProjectFullTextSearchTestCase(APITransactionTestCase):
+    def setUp(self):
+        self.url = reverse("provincial-infrastructure-project-api-list")
+        self.fin_year = FinancialYear.objects.create(slug="2030-31")
+        self.quarter = Quarter.objects.create(number=1)
+        self.date = date(year=2019, month=1, day=1)
+        self.project_1 = ProvInfraProject.objects.create(IRM_project_id=1)
+
+        self.irm_snapshot = IRMSnapshot.objects.create(
+            financial_year=self.fin_year, quarter=self.quarter, date_taken=self.date,
+        )
+        self.project_snapshot_1 = ProvInfraProjectSnapshot.objects.create(
+            irm_snapshot=self.irm_snapshot,
+            project=self.project_1,
+            name="Blue School",
+            province="Eastern Cape",
+            estimated_completion_date=self.date,
+        )
+
+        self.project_2 = ProvInfraProject.objects.create(IRM_project_id=2)
+        self.project_snapshot_2 = ProvInfraProjectSnapshot.objects.create(
+            irm_snapshot=self.irm_snapshot,
+            project=self.project_2,
+            name="Red School",
+            province="Limpopo",
+            estimated_completion_date=self.date,
+        )
+        ProvInfraProjectIndex().reindex()
+
+    def tearDown(self):
+        ProvInfraProjectIndex().clear()
+
+    def test_correct_project_returned(self):
+        search = "Eastern Cape School"
+        data = {"q": search}
+        response = self.client.get(self.url, data)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        results = response.data["results"]
+        self.assertEqual(len(results), 1)
+        self.assertEqual(results[0]["name"], "Blue School")
+        self.assertNotContains(response, "Red School")
+
+
+class ProvInfraProjectDetailPageTestCase(APITransactionTestCase):
+    def setUp(self):
+        self.fin_year = FinancialYear.objects.create(slug="2050-51")
+        self.quarter = Quarter.objects.create(number=3)
+        self.irm_snapshot = IRMSnapshot.objects.create(
+            financial_year=self.fin_year,
+            quarter=self.quarter,
+            date_taken=date(year=2019, month=1, day=1),
+        )
+        self.project = ProvInfraProject.objects.create(IRM_project_id=123456)
+        ProvInfraProjectSnapshot.objects.create(
+            irm_snapshot=self.irm_snapshot,
+            project=self.project,
+            name="Project 123456",
+            department="Health",
+            province="Eastern Cape",
+            status="Construction",
+            primary_funding_source="Health Infrastructure Grant",
+            estimated_completion_date=date(year=2020, month=1, day=1),
+        )
+
+        ProvInfraProjectIndex().reindex()
+
+    def tearDown(self):
+        ProvInfraProjectIndex().clear()
+
+    def test_project_detail_page_fields(self):
```

Comment:
Is this being replaced with the webflow integration selenium tests?

Reply:
yes, here it is:

https://github.com/vulekamali/datamanager/pull/213/commits/c510bce41fc8b2035a3629965510482d78688ad8